src/integration-test/java/com/commercetools/sync/integration/externalsource/producttypes/ProductTypeWithNestedAttributeSyncIT.java

```diff
@@ -245,6 +253,86 @@ void sync_WithNewProductTypeWithANonExistingReference_ShouldCreateProductType()
         });
     }
 
+    @Disabled("Need to investigate since both with missing reference and failed gets counted up? Should it be the case")
+    @Test
+    void sync_WithNewProductTypeWithFailedFetchOnReferenceResolution_ShouldFail() {
+        // preparation
+        final AttributeDefinitionDraft nestedTypeAttr = AttributeDefinitionDraftBuilder
+            .of(AttributeDefinitionBuilder
+                .of("nestedattr", ofEnglish("nestedattr"),
+                    NestedAttributeType.of(ProductType.referenceOfId(PRODUCT_TYPE_KEY_4)))
+                .build())
+            // "isSearchable=true is not supported for attribute type 'nested'."
+            .searchable(false)
+            .build();
+
+        final ProductTypeDraft newProductTypeDraft = ProductTypeDraft.ofAttributeDefinitionDrafts(
+            PRODUCT_TYPE_KEY_1,
+            PRODUCT_TYPE_NAME_1,
+            PRODUCT_TYPE_DESCRIPTION_1,
+            asList(ATTRIBUTE_DEFINITION_DRAFT_1, ATTRIBUTE_DEFINITION_DRAFT_2, nestedTypeAttr));
+
+        final ProductTypeDraft productTypeDraft4 = ProductTypeDraft.ofAttributeDefinitionDrafts(
+            PRODUCT_TYPE_KEY_4,
+            PRODUCT_TYPE_NAME_4,
+            PRODUCT_TYPE_DESCRIPTION_4,
+            singletonList(ATTRIBUTE_DEFINITION_DRAFT_3));
+
+        final SphereClient ctpClient = spy(CTP_TARGET_CLIENT);
+        final BadGatewayException badGatewayException = new BadGatewayException();
+        when(ctpClient.execute(any(ProductTypeQuery.class)))
+            .thenCallRealMethod() // should work on caching
+            .thenCallRealMethod() // should work when fetching matching product types
+            .thenCallRealMethod() // should work when second fetching matching product types
+            .thenReturn(exceptionallyCompletedFuture(badGatewayException)) // fail on fetching during resolution
```

Comment:
a missing reference is not a "fail".. It will be counted as a missing reference and the sync would count it as missing but in this case we want to count it as a fail